### PR TITLE
chore(research): Experiment with TypeScript and Restify for API

### PR DESCRIPTION
**Pull request by jmorgan74**
Created a REST API using TypeScript and Restify too gauge what people think. :-)

## Changes

- Added TypeScript and Restify dependencies.
- Created basic "/hello/:name" route to test the API.

### CODE REVIEW

File: api-microservice/package.json [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "ts-node src/server.ts",
+    "build": "tsc --project ./"
   },
   "keywords": [],
   "author": "",
   "license": "ISC",
```
Comment (MrSimonEmms) [inline api-microservice/package.json:12]: Will probably want to add `"private": true` too

Comment (jmorgan74) [reply api-microservice/package.json:14]: done! :-)

File: api-microservice/package.json [hunk 2]
```diff
@@ -1,15 +1,26 @@
 {
-  "name": "api-microservice",
+  "name": "appeal-planning-decision-api",
   "version": "1.0.0",
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "ts-node src/server.ts",
```
[the remark this answers is not in the document]
Comment (jmorgan74) [reply api-microservice/package.json]: Have added a start:dev and start:prod. start:dev uses nodemon with ts-node and start:prod runs nodemon from the compiles dist directory.

File: api-microservice/package.json [hunk 3]
```diff
@@ -1,15 +1,26 @@
 {
-  "name": "api-microservice",
+  "name": "appeal-planning-decision-api",
   "version": "1.0.0",
```
[the remark this answers is not in the document]
Comment (jmorgan74) [reply api-microservice/package.json]: Done! :-)

File: api-microservice/package.json [hunk 4]
```diff
@@ -1,15 +1,26 @@
...
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "ts-node src/server.ts",
+    "build": "tsc --project ./"
   },
   "keywords": [],
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "@types/restify": "^8.4.2",
+    "express": "^4.17.1",
```
[the remark this answers is not in the document]
Comment (jmorgan74) [reply api-microservice/package.json]: Removed express

**Pull request merged**